wait-for-http/.gitlab-ci.yml

```diff
@@ -0,0 +1,85 @@
+.change_rules_wait_for_http:
+  rules:
+    - if: '$CI_COMMIT_BRANCH == "master"'
+    - &changes_wait_for_http
+      changes:
+        - wait-for-http/**/*
+
+Lint wait-for-http:
+  stage: "Quality"
+  extends: .change_rules_wait_for_http
+  image: hadolint/hadolint:v1.18.2-debian
+  script:
+    - hadolint ./wait-for-http/Dockerfile
+
+Build wait-for-http:
+  stage: "Build"
+  extends:
+    - .base_register_to_gitlab_stage
+    - .change_rules_wait_for_http
+  needs:
+    - Lint wait-for-http
+  variables:
+    CONTEXT: wait-for-http
+    IMAGE_NAME: ${CI_REGISTRY_IMAGE}/wait-for-http
+
+Test wait-for-http 1:
+  needs:
+    - Lint wait-for-http
+    - Build wait-for-http
+  stage: "Test"
+  extends: .change_rules_wait_for_http
+  image:
+    name: "${CI_REGISTRY_IMAGE}/wait-for-http:${CI_COMMIT_SHA}"
+    entrypoint: ["/bin/sh"]
+  variables:
+    RETRIES: 5
+  # expect wait to fail and a retry duration >= 25secs (5*5)
+  script: |
+    start=`date +%s`
+    ! /bin/wait-for-http http://invalid
+    end=`date +%s`
+    runtime=$((end-start))
+    [[ $runtime -gt 25 ]]
+
+# test CLI param usage
+Test wait-for-http 2:
+  needs:
+    - Lint wait-for-http
+    - Build wait-for-http
+  stage: "Test"
+  extends: .change_rules_wait_for_http
+  image:
+    name: "${CI_REGISTRY_IMAGE}/wait-for-http:${CI_COMMIT_SHA}"
+    entrypoint: ["/bin/sh"]
+  script:
+    - /bin/wait-for-http http://free.fr
+
+# test URL env variable usage
+Test wait-for-http 3:
+  needs:
+    - Lint wait-for-http
+    - Build wait-for-http
+  stage: "Test"
+  extends: .change_rules_wait_for_http
+  image:
+    name: "${CI_REGISTRY_IMAGE}/wait-for-http:${CI_COMMIT_SHA}"
+    entrypoint: ["/bin/sh"]
+  variables:
+    URL: http://free.fr
+  script:
+    - /bin/wait-for-http
```

Comment:
```suggestion
  script:
    - /bin/wait-for-http http://free.fr
```

Reply:
the first test ensure it fallback to ENV var when the 1st arg is not given